### ISSUE TITLE
Tests hang when collecting coverage

### ISSUE THREAD

**Issue by Hixie** [flutter/flutter#7535]
If your `flutter/test` directory contains only the following files:
```
rendering_tester.dart
slivers_test.dart
stack_test.dart
table_test.dart
transform_test.dart
viewport_test.dart
```
...and you run `flutter test --coverage`, it will hang after running 14 tests and skipping one.

Removing any of the above files removes the hang. Removing `--coverage` removes the hang.

**Comment by Hixie** [comment]
Replacing `viewport_test.dart` witha file consisting of only `void main() { }` doesn't remove the problem.

**Comment by Hixie** [comment]
I have it reduced to five files named `a_test.dart` .. `e_test.dart`, each containing nothing more than `void main() { }`.

**Comment by Hixie** [comment]
It just seems to be a matter of how many test files there are. Any odd number of tests greater or equal to three, it hangs.

**Comment by abarth** [comment]
I thought I had a fix for this issue, but when I cleaned it up, it didn't fix the issue anymore.

**Comment by Hixie** [comment]
Hopefully fixed by https://github.com/flutter/flutter/pull/7548

**Comment by Hixie** [comment]
Not that we'll know since we're at an even number of tests...

**Comment by cbracken** [comment]
@Hixie Is this still an issue? IIRC this was distinct from dart-lang/coverage#165?

**Comment by Hixie** [comment]
Yeah should have been fixed by #7548.
